encodings/fastlanes/src/bitpacking/serde.rs

```diff
@@ -112,11 +114,20 @@ impl EncodingVTable for BitPackedEncoding {
             None => find_best_bit_width(&parray)?,
         };
 
-        if bit_width as usize == parray.ptype().bit_width() {
-            return Ok(Some(parray.to_array()));
-        }
-
-        Ok(Some(bitpack_encode(&parray, bit_width)?.into_array()))
+        Ok(Some(
+            match (bit_width as usize).cmp(&parray.ptype().bit_width()) {
+                Ordering::Less => bitpack_encode(&parray, bit_width)?.into_array(),
+                Ordering::Equal => parray.to_array(),
+                Ordering::Greater => {
+                    let new_bit_width = find_best_bit_width(&parray)?;
+                    if new_bit_width as usize >= parray.ptype().bit_width() {
+                        parray.to_array()
+                    } else {
+                        bitpack_encode(&parray, new_bit_width)?.into_array()
+                    }
+                }
+            },
+        ))
```

Comment:
Isn't the logic you're adding only applicable in the case when there's a like argument? I think we can split this and move part of it to the present branch of like and keep a line or two here for actual encode logic